xdsl/passes.py

```diff
@@ -151,23 +159,30 @@ def pipeline_pass_spec(self, *, include_default: bool = False) -> PipelinePassSp
         return PipelinePassSpec(self.name, args)
 
 
-# Git Issue: https://github.com/xdslproject/xdsl/issues/1845
-def get_pass_argument_names_and_types(arg: type[ModulePassT]) -> str:
+class PassArgOption(NamedTuple):
```

Comment:
```suggestion
class PassOption(NamedTuple):
```

I know we use "arg", but the "LLVM lingo" is option (e.g., `cl::opt<>` from `CommandLine.h`).
Maybe let's start using it like that and we can rename the other parts later (e.g., `get_pass_option_info`).

Reply:
Or even `PassOptionInfo`? I feel it leaves room to grow.

Reply:
I like PassOptionInfo more as it's more the information about what is expected, rather than what is passed.